File: order-service/src/main/java/com/cg/orderservice/services/OrderService.java
/**
 * @author Gagandeep Singh
 * @email singh.gagandeep3911@gmail.com
 * @create date 2021-01-14 01:05:18
 * @modify date 2021-01-14 01:05:18
 * @desc [description]
 */
package com.cg.orderservice.services;

import java.util.List;

import com.cg.orderservice.dto.UpdateStatusDto;
import com.cg.orderservice.entities.OrderMain;

public interface OrderService {

  List<OrderMain> fetchByUserId(Long userId);

  List<OrderMain> fetchAll();

  OrderMain findByOrderId(Long orderId);

  OrderMain updateOrderStatus(UpdateStatusDto updateStatusDto);

}

File: frontend/src/app/services/order.service.ts
/**
 * @author Gagandeep Singh
 * @email singh.gagandeep3911@gmail.com
 * @create date 2021-01-26 11:35:44
 * @modify date 2021-01-26 11:35:44
 * @desc [description]
 */
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { UpdateOrderStatus } from '../models/update-order-status.model';
import { AuthService } from './auth.service';

@Injectable({
  providedIn: 'root',
})
export class OrderService {
  orderServiceUrl = `${environment.protocol}${environment.applicationUrl}/${environment.orderService}/orders`;

  constructor(private http: HttpClient, private authService: AuthService) {}

  updateOrderStatus(data: UpdateOrderStatus) {
    return this.http.post(`${this.orderServiceUrl}`, data);
  }

  fetchOrder() {
    return this.http.get(`${this.orderServiceUrl}`);
  }

  fetchById(id) {
    return this.http.get(`${this.orderServiceUrl}/${id}`);
  }

  fetchByLoggedInUserId() {
    return this.fetchByUserId(
      this.authService.fetchFromSessionStorage()?.userId
    );
  }

  fetchByUserId(userId) {
    return this.http.get(`${this.orderServiceUrl}/user/${userId}`);
  }
}
